apps/frontend/src/libs/components/create-route-form/create-route-form.tsx

```diff
@@ -0,0 +1,203 @@
+
+	useEffect(() => {
+		if (
+			previousCreateStatusReference.current === DataStatus.PENDING &&
+			createStatus === DataStatus.FULFILLED
+		) {
+			closeModal?.();
+		}
+
+		previousCreateStatusReference.current = createStatus;
+	}, [createStatus, closeModal]);
+
+	useEffect(() => {
+		const loadSavedData = async (): Promise<void> => {
+			const savedData = await storage.get(StorageKey.CREATE_ROUTE_FORM_DATA);
+
+			if (!savedData) {
+				return;
+			}
+
+			try {
+				const { description, name } = JSON.parse(savedData) as {
+					description: string;
+					name: string;
+				};
+
+				if (name) {
+					handleValueSet("name", name);
+				}
+
+				if (description) {
+					handleValueSet("description", description);
+				}
+			} catch {
+				await storage.drop(StorageKey.CREATE_ROUTE_FORM_DATA);
+			}
+		};
+
+		void loadSavedData();
+	}, [handleValueSet]);
+
+	useEffect(() => {
+		const subscription = control._subjects.state.subscribe({
+			next: () => {
+				const { description = "", name = "" } = control._formValues as {
+					description: string;
+					name: string;
+				};
+
+				if (name || description) {
+					void storage.set(
+						StorageKey.CREATE_ROUTE_FORM_DATA,
+						JSON.stringify({ description, name }),
+					);
+				} else {
+					void storage.drop(StorageKey.CREATE_ROUTE_FORM_DATA);
+				}
+			},
+		});
+
+		return (): void => {
+			subscription.unsubscribe();
+		};
+	}, [control]);
+
+	const handleModalClose = useCallback((): void => {
+		if (!hasBeenSubmittedReference.current) {
+			void storage.drop(StorageKey.CREATE_ROUTE_FORM_DATA);
+		}
+	}, []);
+
+	useEffect((): void => {
+		registerCleanup?.(handleModalClose);
+	}, [registerCleanup, handleModalClose]);
+
+	const handleSubmit = useCallback(
+		(data: typeof DEFAULT_CREATE_ROUTE_PAYLOAD): void => {
+			if (!onSubmit) {
+				return;
+			}
+
+			const plannedRouteId = searchParameters.get("plannedRouteId");
+
+			const completeData: RouteCreateRequestDto = {
+				...data,
+				createdByUserId: authenticatedUser.id,
+				plannedPathId: plannedRouteId ? Number(plannedRouteId) : 0,
+			};
+
+			hasBeenSubmittedReference.current = true;
+			onSubmit(completeData);
+			void storage.drop(StorageKey.CREATE_ROUTE_FORM_DATA);
+		},
+		[onSubmit, authenticatedUser.id, searchParameters],
+	);
+
+	return (
+		<div>
+			<h4 className={styles["title"]}>Create New Route</h4>
+			<form
+				className={styles["form"]}
+				onSubmit={handleFormSubmit(handleSubmit)}
+			>
+				<div className={styles["field"]}>
+					<Input control={control} errors={errors} label="Name" name="name" />
+				</div>
+
+				<div className={styles["field"]}>
+					<div className={styles["route-section"]}>
+						<span className={styles["label"]}>Route</span>
+						{isRouteConstructed && (
+							<div>
+								<span>Route constructed</span>
+							</div>
+						)}
+						<div>
+							<Button
+								label="Construct"
+								to="/routes/construct?returnTo=/routes?modal=create-route"
```

Comment:
I have a feeling that this should be a template string with enums. Something like this maybe:

```suggestion
	to=`${AppRoute.ROUTES_CONSTRUCT}?returnTo=${AppRoute.ROUTES}?modal=${ModalType.CREATE_ROUTE}"
```
But I'm not completely sure if we need this here. And, also there is this "/app" in these enums. What do you think?